docker/prod/entrypoint.sh

```diff
@@ -30,8 +30,19 @@ if [ "$(id -u)" = '0' ]; then
 	# reexport PGVERSION and PGBIN env variables according to postgres version of existing cluster (if any)
 	# this must happen in the entrypoint
 	if [ -f "$PGDATA/PG_VERSION" ]; then
-		export PGVERSION="$(cat "$PGDATA/PG_VERSION")"
+		EXISTING_PGVERSION="$(cat "$PGDATA/PG_VERSION")"
 		echo "-----> Existing PostgreSQL cluster found in $PGDATA."
+
+		# Check for version mismatch: if Docker default is PG 17 but existing data is PG 13
+		if [ "$PGVERSION" = "17" ] && [ "$EXISTING_PGVERSION" = "13" ]; then
```

Comment:
@oliverguenther Should we perhaps not make this specific to 17 and 13 but show this warning in general if there is a mismatch between the set PGVERSION and the one found in the cluster?

I know coming from our own setups, this should be the only combination really, but it probably doesn't hurt to catch more potential issues here.